src/core.js

```diff
@@ -49,7 +49,13 @@ function createModule( name, testEnvironment ) {
 		testsRun: 0,
 		unskippedTestsRun: 0,
 		childModules: [],
-		suiteReport: new SuiteReport( name, parentSuite )
+		suiteReport: new SuiteReport( name, parentSuite ),
+
+		// Pass along `skip` property from parent module, in case there is one,
+		// to childs. And use own otherwise.
+		// This property will be used to mark own tests and tests of child suites
+		// as `skipped`.
+		skip: parentModule !== null && parentModule.skip || modifiers.skip
```

Comment:
I don't think the order here matters. If the module is skipped, then it'll be skipped. If the parent is skipped, it will also be skipped. There is no way to "unskip".

I agree with the rest of your implementation plan :+1:.